store/collection_operations.go

```diff
@@ -30,6 +31,17 @@ func (store *Store) UpdateCollectionID(ctx context.Context, path, collectionID s
 		return ErrCollectionIDAlreadySet
 	}
 
+	//check to see if collectionID exists and is not-published
+	m := files.StoredRegisteredMetaData{}
+	if err := store.mongoCollection.FindOne(ctx, bson.M{fieldCollectionID: collectionID}, &m); err != nil && !errors.Is(err, mongodriver.ErrNoDocumentFound) {
+		log.Error(ctx, "update collection ID: caught db error", err, logdata)
+		return err
+	}
+	if m.State == StatePublished || m.State == StateDecrypted {
```

Comment:
minor - but we could refactor this to a method of `StoredRegisteredMetaData` - just thinking...we have defined this twice so if we want to change would need to change twice. but with method - just one place.